src/txpasslib/test/matchers.py

```diff
@@ -0,0 +1,20 @@
+from operator import attrgetter
+
+from testtools.matchers import AfterPreprocessing, Equals, MatchesAll
+from testtools.twistedsupport import failed
+
+
+def failed_with(matcher):
+    """
+    Match against the exception of a failure.
+    """
+    return failed(AfterPreprocessing(attrgetter('value'), matcher))
+
+
+def performed(perform, matcher):
```

Comment:
Hmm, well, `perform` is basically going to show up anywhere we have a thread pool/worker; I guess the question is, is that going to be anywhere besides `txpasslib.context`? I figure the answer is "yes". I'll add the docstring, though.